flink-table/flink-table-common/src/main/java/org/apache/flink/table/catalog/ReadableWritableCatalog.java

```diff
@@ -0,0 +1,154 @@
+	 * @param ignoreIfNotExists Flag to specify behavior when the database does not exist:
+	 *                          if set to false, throw an exception,
+	 *                          if set to true, do nothing.
+	 * @throws DatabaseNotExistException if the given database does not exist
+	 */
+	void dropDatabase(String name, boolean ignoreIfNotExists) throws DatabaseNotExistException;
+
+	/**
+	 * Modify an existing database.
+	 *
+	 * @param name        Name of the database to be modified
+	 * @param newDatabase    The new database definition
+	 * @param ignoreIfNotExists Flag to specify behavior when the given database does not exist:
+	 *                          if set to false, throw an exception,
+	 *                          if set to true, do nothing.
+	 * @throws DatabaseNotExistException if the given database does not exist
+	 */
+	void alterDatabase(String name, CatalogDatabase newDatabase, boolean ignoreIfNotExists)
+		throws DatabaseNotExistException;
+
+	// ------ tables and views ------
+
+	/**
+	 * Drop a table or view.
+	 *
+	 * @param tablePath         Path of the table or view to be dropped
+	 * @param ignoreIfNotExists Flag to specify behavior when the table or view does not exist:
+	 *                          if set to false, throw an exception,
+	 *                          if set to true, do nothing.
+	 * @throws TableNotExistException if the table or view does not exist
+	 */
+	void dropTable(ObjectPath tablePath, boolean ignoreIfNotExists) throws TableNotExistException;
+
+	/**
+	 * Rename an existing table or view.
+	 *
+	 * @param tablePath       Path of the table or view to rename
+	 * @param newTableName     the new name of the table or view
+	 * @param ignoreIfNotExists Flag to specify behavior when the table or view does not exist:
+	 *                          if set to false, throw an exception,
+	 *                          if set to true, do nothing.
+	 * @throws TableNotExistException if the table does not exist
+	 * @throws DatabaseNotExistException if the database in tablePath to doesn't exist
+	 */
+	void renameTable(ObjectPath tablePath, String newTableName, boolean ignoreIfNotExists)
+		throws TableNotExistException, DatabaseNotExistException;
+
+	// ------ tables ------
+
+	/**
+	 * Create a new table. Note that TableStats in the table is ignored for table creation.
+	 *
+	 * @param tablePath      Path of the table to be created
+	 * @param table          The table definition
+	 * @param ignoreIfExists Flag to specify behavior when a table already exists at the given path:
+	 *                       if set to false, it throws a TableAlreadyExistException,
+	 *                       if set to true, do nothing.
+	 * @throws TableAlreadyExistException if table already exists and ignoreIfExists is false
+	 * @throws DatabaseNotExistException if the database in tablePath doesn't exist
+	 */
+	void createTable(ObjectPath tablePath, CatalogTable table, boolean ignoreIfExists)
```

Comment:
It seems table in these APIs can be either table&view or table. For example, the `Table` in `getTable, renameTable, dropTable` means table and view, while the `Table` in `createTable, alterTable` means only table. It seems not very clear.

I think we have two options to solve the problem.
1.treat table as table and view for all APIs
2.treat table as the only table for all APIs. 
However, the first option may be more in line with the behavior of most standard databases.

Reply:
I think the two set of APIs are there for a reason.
1. getTable, renameTable, and dropTable only takes a name and nothing specific to either table or view.
2. on the contrary, createTable/View and alterTable/View takes a CatalogTable instance or  a CataView instance, and the implementations are expected to be different.

While it might seem confusing, it makes good sense if we look at the method signatures. Let me know if you still think this is confusing.

Reply:
This involves an important concept, whether the Table contains (table and view), and if so, then I still recommend to keep a set of createTable/dropTable/renameTable/alterTable APIs, remove createView/alterView. any thoughts.?

Reply:
Well, createView/alterView takes different input than createTable/alterTable. If we mingle them, then the implementation of the API will be nasty because it will have to do something like if commonTable instance of CatalogTable then do xxx else if commonTable instanceof CatalogView, then do yyy. I'm not sure if we need a parity between view and table. There are things to be shared. let me go back to check how it's done in spark and presto to get some ideas there.

Reply:
Sorry to jump into the discussion. Maybe it would be better if we can tell the differences directly and clearly from the method name. 

Currently, the method names are not trusted and we have to look at the signature to tell the differences. However, when we looking at the `List<String> listTables(String dbName)` method, it would again bring confusions. From the signature, a user will think both tables and views will be listed while actually only tables but not views are listed. 

I agree that the implementations of createTable and creatView are expected to be different, but I think it's ok to use one api for both of them. Take inmemory catalog as an example, we even don't need the `private helper method`. And we can use the `a private helper method` to solve the problem if the implementations are different. 

What do you guys think?

Best, 
Hequn

Reply:
Thanks all for the feedback. I don't think either method name or signature of the method or both can tell the whole story. That's the reason why we have API document, which is indispensable for API users. If one is still confused even after looking at the API documentation, then we have a problem. However, I don't think it's the case here. Plus, naming is very subjective and it can never be perfect!

Nevertheless, I take the point of merging APIs for regular tables and views into a single set, which is reflected in the latest update.

Let me know if there is anything to be further changed. Thanks.